python/lsst/ts/externalscripts/make_love_stress_tests.py

```diff
@@ -0,0 +1,318 @@
+        """Configure the script.
+
+        Specify the Stress test configurations:
+        - LOVE host location
+        - Number of clients
+        - Number of messages to store
+        - CSCs
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration with several attributes:
+
+            * location : a string, with the location of the
+                LOVE instance to stress
+            * number_of_clients : a number that indicates the number
+                of clients to create
+            * number_of_messages : a number that indicates the number
+                of messages to store before calculating the mean latency
+            * data : a list, where each element is a string in the form:
+
+                * CSC name and optional index as ``csc_name:index`` (a `str`).
+
+        Notes
+        -----
+        Saves the results on several attributes:
+
+        * location : a string, with the location of the LOVE instance to stress
+        * number_of_clients : a number, with the number of clients to create
+        * number_of_messages : a number, with the number of messages to store
+        * remotes : a dict of (csc_name, index): remote,
+            an `lsst.ts.salobj.Remote`
+
+        Constructing a `salobj.Remote` is slow (DM-17904), so configuration
+        may take a 10s or 100s of seconds per CSC.
+        """
+        self.log.info("Configure started")
+
+        # set configurations
+        self.location = config.location
+        self.number_of_clients = config.number_of_clients
+        self.number_of_messages = config.number_of_messages
+
+        # get credentials
+        self.username = "user"
+        self.password = os.environ.get("USER_USER_PASS")
+
+        if self.password is None:
+            raise Exception("Configuration failed")
+
+        # construct remotes
+        remotes = dict()
+        for elt in config.data:
+            name, index = salobj.name_to_name_index(elt)
+            self.log.debug(f"Create remote {name}:{index}")
+            if elt not in remotes:
+                remote = salobj.Remote(domain=self.domain, name=name, index=index)
+                remotes[elt] = remote
+
+        self.remotes = remotes
+
+    async def run(self):
+        """Run script."""
+        tasks = [
+            remote.start_task
+            for remote in self.remotes.values()
+            if not remote.start_task.done()
+        ]
+        if tasks:
+            self.log.info(f"Waiting for {len(tasks)} remotes to be ready")
+            await asyncio.gather(*tasks)
+
+        # Enable all CSCs
+        for csc, remote in self.remotes.items():
+            await salobj.set_summary_state(remote=remote, state=salobj.State.ENABLED)
+
+        event_streams = dict()
+        telemetry_streams = dict()
+        for producer in self.remotes.keys():
+            event_streams[producer] = salobj.SalInfo(
+                self.domain, producer.split(":")[0]
```

Comment:
Please do NOT create multiple SalInfo. There is no point, and it is expensive. Also this use of getattr is confusing, because you can just get the attribute directly.

Also please use name_to_name_index everywhere. Just look for all instances of `split` in this code.

Consider something like this:
```
for name_index in self.remotes:  # .keys() is optional
    name, index = salobj.name_to_name_index(name_index)
    salinfo = salobj.SalInfo(domain=self.domain, name=name)
    try:
        event_streams[name_index] = salinfo.event_names
        telemetry_streams[name_index] = salinfo.telemetry_names
    finally:
        await salinfo.close()
```
Finally I strongly suggest `name_index` as the variable name for a SAL component name with index. I see at least 3 different variable names in this code: `elt`, `csc`, and `producer`, all of which mean the same thing. `elt` is uninformative, and `csc` and `producer` sound like something much larger (e.g. an instance of salobj.BaseCsc or some class named Producer).